code/Model/EditableFormField.php

```diff
@@ -321,6 +323,17 @@ public function getCMSFields()
         return $fields;
     }
 
+
+    public function requireDefaultRecords()
+    {
+        parent::requireDefaultRecords();
+
+        // make sure to migrate the class across (prior to v5.x)
+        DB::query("UPDATE EditableFormField SET ParentClass = 'Page' WHERE ParentClass IS NULL");
+        DB::query("UPDATE EditableFormField_Live SET ParentClass = 'Page' WHERE ParentClass IS NULL");
+        DB::query("UPDATE EditableFormField_Versions SET ParentClass = 'Page' WHERE ParentClass IS NULL");
```

Comment:
I think this is what the legacy class remapping feature does, or is that not the case?

Reply:
I changed the has_one to DataObject which requires an explicit class name.

Reply:
Ah - right. Ok, so this code will run on every dev/build for everyone using userforms in SS4 - maybe we could move it into a userforms 5.x upgrade task instead - thoughts?

Reply:
Resolved IRL - we'll leave this as it is to ensure it gets run, rather than expect people to remember to run the manual task. It shouldn't cause a maintenance burden going forward.